query/src/users/auth/auth_mgr.rs

```diff
@@ -64,6 +68,15 @@ impl AuthMgr {
                     .tenant_id
                     .clone()
                     .unwrap_or_else(|| self.tenant.clone());
+                let identity = UserIdentity {
+                    username: user_name.to_string(),
+                    hostname: h.as_ref().unwrap_or(&"%".to_string()).to_string(),
+                };
+                if identity.is_root() && !identity.is_localhost() {
+                    return Err(ErrorCode::AuthenticateFailure(
+                        "root user can only login from localhost",
+                    ));
+                }
```

Comment:
We should try to avoid special logic, if user 'root'@'other_host' is not found, it will not be able to login